Core/Base/ConsumerReactor.py

```diff
@@ -0,0 +1,50 @@
+"""
+  DIRAC class to execute consumers
+
+  All DIRAC consumers must inherit from the basic class ConsumerModule
+
+  The class is created based on the AgentReactor example.
+  In the most common case, DIRAC consumers are executed using the dirac-consumer command.
+  dirac-consumer accepts a list positional arguments. These arguments have the form:
+  [DIRAC System Name]/[DIRAC Consumer Name]
+  dirac-consumer then:
+  - produces a instance of ConsumerReactor
+  - loads the required modules using the ConsumerReactor.loadConsumerModules method
+  - starts the execution loop using the ConsumerReactor.go method
+
+"""
+
+from DIRAC.Core.Base.ConsumerModule import ConsumerModule
+from DIRAC.Core.Base.private.ModuleLoader import ModuleLoader
+from DIRAC.ConfigurationSystem.Client import PathFinder
+from DIRAC import S_OK
+
+
+class ConsumerReactor(object):
+  """
+    Main interface to DIRAC consumers.
+  """
+
+  def __init__( self ):
+    self.__loader = ModuleLoader( "Consumer", PathFinder.getConsumerSection, ConsumerModule )
+    self.__consumerModules={}
+
+  def go( self ):
+    """
+      Main method to control the execution of all configured consumers
+    """
+    for name in self.__consumerModules:
```

Comment:
Just a naive question for my education. Is it possible to get several modules to execute ? In which case, according to this loop, they would just be executed sequentially ? Do you have an example of such a case ?